src/full/Agda/Interaction/MakeCase.hs

```diff
@@ -303,6 +310,12 @@ makeCase hole rng s = withInteractionId hole $ do
         ifM (liftTCM $ isEmptyTel (scTel sc))
           {- then -} (pure Nothing)
           {- else -} (Just <$> makeAbstractClause f rhs sc)
+    -- 3. If the definition is left empty then rewind and insert a single absurd clause
+    cs <- if not (null cs) || not (null otherClauses) then pure cs else do
```

Comment:
Isn't the right thing to do to only look at the new clauses generated by the case split, and check that at least one of them is kept by the filtering? So you wouldn't need to look at the `otherClauses` at all.

Reply:
If you have

```agda
both' : ∀ (c : Bool) b → b ≡ true → b ≡ false → Bool
both' false b e f = {!!}
both' c b e f = {!!}
```

then dropping `both' false b e f = {!!}` upon splitting on `e` is perfectly fine.
But you are right that we only have an obligation to generate valid code, not the
most concise code possible no matter what splitting strategy the user chooses.